### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by willbeason**
Use go get -u to upgrade dependencies.

Related to: https://github.com/open-policy-agent/gatekeeper/pull/1335

The big thing here is upgrading opa from v0.24.0 to v0.29.4.

We are unable to safely upgrade to k8s 1.21 libraries until controller-runtime has a stable v0.9.0 release. That can be done later.

Remove installing opa v0.10 directly in the docker image as it seems vestigial and is no longer required.

Signed-off-by: Will Beason <willbeason@google.com>

### CODE REVIEW

**Review by maxsmythe — APPROVED** [review]
LGTM

@sozercan @ritazh @shomron LGTY?

File: constraint/config/crds/templates.gatekeeper.sh_constrainttemplates.yaml [hunk 1]
```diff
@@ -19,18 +19,13 @@ spec:
   - name: v1alpha1
     schema:
       openAPIV3Schema:
-        description: ConstraintTemplate is the Schema for the constrainttemplates
-          API
+        description: ConstraintTemplate is the Schema for the constrainttemplates API
```
Comment (ritazh) [inline constraint/config/crds/templates.gatekeeper.sh_constrainttemplates.yaml:22]: @julianKatz this is the same line break I'm seeing as well

File: constraint/Dockerfile [hunk 2]
```diff
@@ -21,11 +21,6 @@ RUN curl -L -O "https://github.com/kubernetes-sigs/kustomize/releases/download/v
     mv kustomize_${version}_linux_${arch} /usr/bin/kustomize &&\
     chmod u+x /usr/bin/kustomize
 
-# Install OPA
-WORKDIR /usr/bin
-RUN curl -L -o opa https://github.com/open-policy-agent/opa/releases/download/v0.10.5/opa_linux_amd64 &&\
-    chmod 755 opa
```
Comment (ritazh) [inline constraint/Dockerfile:27]: 👍

**Comment by ritazh** [comment]
unit test failed:
> package sigs.k8s.io/controller-tools/cmd/controller-gen@v0.5.0: can only use path@version syntax with 'go get'
make: *** [Makefile:34: gen-dependencies] Error 1

File: constraint/config/crds/templates.gatekeeper.sh_constrainttemplates.yaml (hunk 1 above)
Comment (maxsmythe) [reply constraint/config/crds/templates.gatekeeper.sh_constrainttemplates.yaml:22]: I wonder if this is version skew against the version of controller-gen used. Did we dockerize that here?

Comment (maxsmythe) [reply constraint/config/crds/templates.gatekeeper.sh_constrainttemplates.yaml:22]: Or maybe kustomize

Comment (willbeason) [reply constraint/config/crds/templates.gatekeeper.sh_constrainttemplates.yaml:22]: > unit test failed:
> 
> > package sigs.k8s.io/controller-tools/cmd/controller-gen@v0.5.0: can only use path@version syntax with 'go get'
> > make: *** [Makefile:34: gen-dependencies] Error 1

Ah, I found it.

**Comment by willbeason** [comment]
> unit test failed:
> 
> > package sigs.k8s.io/controller-tools/cmd/controller-gen@v0.5.0: can only use path@version syntax with 'go get'
> > make: *** [Makefile:34: gen-dependencies] Error 1

Ah, I found it.

**Comment by sozercan** [comment]
@willbeason were you able to find the cause of unit test failure?

**Comment by willbeason** [comment]
> @willbeason were you able to find the cause of unit test failure?

Ah, I see. The GitHub Action was setting up Go 1.15 but we need 1.16. Fixed.

**Comment by willbeason** [comment]
Using `git lfs` to track the large "*.a" files.

**Review by sozercan — APPROVED** [review]
LGTM

**Pull request merged**